### ISSUE TITLE
Cannot put a breakpoint on the last line of a function if it's a complex boolean expression

### ISSUE THREAD

**Issue by asik** [dotnet/fsharp#1390]
I noticed this while writing property tests with FsCheck.Xunit.
Let's say you're writing a simple property test like this:

```
[<Property>]
let fact seed =
    let rng = Random seed
    let a = rng.Next() % 2 = 0
    let b = rng.Next() % 2 = 0
    a && b
```

You cannot step or put a breakpoint on the last line (which would be nice, because you'd like to know for which reason the test is failing). This is especially annoying because Visual Studio will let you put the breakpoint, but then when you start debugging, strange things can happen like the debugger never actually breaking inside your code, or in my case, the test runner [failing with cryptic errors](http://stackoverflow.com/questions/38647311/unauthorizedaccessexception-on-pdb-trying-to-debug-a-unit-test/38655065). Wasted a lot of time figuring out what was actually going on yesterday.

A simple workaround is to bind the expression to a local and return that instead:

```
[<Property>]
let fact seed  =
    let rng = Random seed
    let a = rng.Next() % 2 = 0
    let b = rng.Next() % 2 = 0
    let result = a && b
    result
```

But I think the the compiler should do the work for me.

Tested on Visual Studio 2015 Enterprise Update 3.

**Comment by cartermp** [comment]
Can confirm that this is still happening.

```fsharp
let and' x y =
    x && y

[<EntryPoint>]
let main argv = 
    and' true false |> ignore
    0 // return an integer exit code
```
The breakpoint will resolve on the `x && y` line, but when debugging, it jumps to the function call, and you cannot step into the call.

**Comment by JordanMarr** [comment]
I was going to file this exact bug as I just rediscovered it yesterday in VS 2019. I also used the same workaround of creating a temp binding to allow me to create the breakpoint.

**Comment by dsyme** [comment]
Fixed in #11981
